buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -225,21 +225,28 @@ private void allocate(PoolThreadCache cache, PooledByteBuf<T> buf, final int req
         }
     }
 
-    private synchronized void allocateNormal(PooledByteBuf<T> buf, int reqCapacity, int normCapacity) {
-        if (q050.allocate(buf, reqCapacity, normCapacity) || q025.allocate(buf, reqCapacity, normCapacity) ||
-            q000.allocate(buf, reqCapacity, normCapacity) || qInit.allocate(buf, reqCapacity, normCapacity) ||
-            q075.allocate(buf, reqCapacity, normCapacity)) {
-            ++allocationsNormal;
-            return;
-        }
+    private void allocateNormal(PooledByteBuf<T> buf, int reqCapacity, int normCapacity) {
+        synchronized (this) {
```

Comment:
Just for my understanding, you removed ```synchronized``` from the method declaration and introduced this ```synchronized block```. Is this because you realized that the call to ```allocateHuge()``` later in this method is not required to be within a ```synchronized``` portion of code?